scrapper.py

```diff
@@ -25,39 +36,159 @@ class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls, max_articles: int):
-        pass
 
-    def _extract_url(self, article_bs):
-        pass
+    def __init__(self, seed_urls, max_articles: int):
+        self.max_articles = max_articles
+        self.seed_urls = seed_urls
+        self.count_articles = 0
+        self.urls = []
+
+    def _extract_url(self, article_bs):  # I don't understand if I need to collect seed urls (from main to year)
+        # or just put it in scrapper_config.
+        articles_code_stored_here = article_bs.find_all('div', {'class': 'articles'})
+        for article_code in articles_code_stored_here:
+            links_articles = article_code.find_all('a')
+            for link in links_articles:
+                if self.max_articles == self.count_articles:
+                    break
+                try:
+                    art_link = link["href"]
+                    if art_link.startswith('/') and 'pdf' not in art_link:
+                        self.urls.append('https://periodical.pstgu.ru' + link["href"])
+                        self.count_articles += 1
+                except KeyError:
+                    continue
 
     def find_articles(self):
         """
         Finds articles
-        """
-        pass
+        """  # I also don't understand what I need to do here, because this code can be moved to extract_url
+        for seed_url in self.seed_urls:
+            issue_page = requests.get(seed_url)
+            issue_page_bs = BeautifulSoup(issue_page.text, 'html.parser')
+            self._extract_url(issue_page_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def parse(self):
+        article_page = requests.get(self.url)
+        article_bs = BeautifulSoup(article_page.text, 'html.parser')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        list_of_possible_l = ['Список литературы', 'ЛИТЕРАТУРА', 'Список использованных источников и литературы',
+                              'Источники']
+
+        class_of_link = article_bs.find('div', {'class': "articles-item-more"})
+        if class_of_link:
+            link_to_save = class_of_link.find('a')
+            link_to_pdf = 'https://periodical.pstgu.ru' + link_to_save['href']
```

Comment:
do you see - you again use that constant! another reason to think of central place for defining it